README.md

```diff
@@ -15,18 +15,48 @@ Just doing `stack install yi` would be akin to unwrapping a box of legos and
 finding an assembled spaceship there.
 Note that this way, it will not be possible to use a custom configuration.
 
-In order to have a personalized configuration, it is required to use yi as libraries to create your own text editor.
-
-A good starting point is choosing an example configuration of your liking in the
-[example-configs][userconfigs] directory, building it, running it, and tinkering with it.
+In order to have a personalized configuration, it is required to use Yi as libraries to create your own text editor.
 
 See [this documentation page](https://yi-editor.github.io/pages/installing/)
 for more detailed installation instructions. Hacking instructions if you're
 using the nix package manager are also there.
 
-### Advanced configuration examples
+## Configuring
+
+A good starting point is choosing an example configuration of your liking in the
+[example-configs][userconfigs] directory, building it, running it, and tinkering with it.
 
-Some people share their yi configuration here on [github](https://github.com/search?utf8=%E2%9C%93&q=yi-config+language%3Ahaskell&type=).
+Yi, as a library, can be categorized into four parts :
+
+* *UIs*, how the editor is rendered following input events;
```

Comment:
In the section title on line 24: "Configuration" seems more appropriate than "Configuring"